conda_build/render.py

```diff
@@ -860,6 +869,11 @@ def distribute_variants(
                 )
                 or mv.config.variants
             )
+        # copy variants before we start modifying them,
+        # but after we've reduced the list via the conform_dict filter
+        mv.config.variants = pickle.loads(
+            pickle.dumps(mv.config.variants, pickle.HIGHEST_PROTOCOL)
+        )
```

Comment:
Should we just add a `copy_variants` method to config to reduce doing the same thing as https://github.com/conda/conda-build/blob/adfdc2a43d968c934d14bbaa30b1a023f68a77fc/conda_build/config.py#L827-L829 ?

Reply:
good idea, added.